node/service/src/grandpa_support.rs

```diff
@@ -123,16 +131,16 @@ impl<B> grandpa::VotingRule<PolkadotBlock, B> for ApprovalCheckingDiagnostic
 		let mut overseer = self.overseer.clone();
 		let checking_lag = self.checking_lag.clone();
 
-		let current_hash = current_target.hash();
-		let current_number = current_target.number.clone();
+		let best_hash = best_target.hash();
+		let best_number = best_target.number.clone();
 
 		let base_number = base.number;
 
 		Box::pin(async move {
 			let (tx, rx) = oneshot::channel();
 			let approval_checking_subsystem_vote = {
 				overseer.send_msg(ApprovalVotingMessage::ApprovedAncestor(
-					current_hash,
+					best_hash,
```

Comment:
nit: ApprovedAncestor also needs `best_number`, we can provide it here to avoid an extra `ChainApi` call

Reply:
While true, beyond the scope here